src/merge.rs

```diff
@@ -0,0 +1,138 @@
+use std::marker;
+use std::mem;
+use libc::c_uint;
+
+use {raw, Object, FileFavor};
+use util::Binding;
+use call::Convert;
+
+/// annotated commits, the input to merge and rebase
+pub struct AnnotatedCommit<'repo> {
+    raw: *mut raw::git_annotated_commit,
+    _marker: marker::PhantomData<Object<'repo>>,
```

Comment:
Looking at the definition in C, it looks like this actually contains a `Commit`, so could this phantom data be `Commit<'repo>` instead?